zerver/lib/message.py

```diff
@@ -1190,9 +1196,13 @@ def get_huddle_users(recipient_id: int) -> str:
         # TODO: Add support for alert words here as well.
         is_mentioned = (row["flags"] & UserMessage.flags.mentioned) != 0
         is_wildcard_mentioned = (row["flags"] & UserMessage.flags.wildcard_mentioned) != 0
+        is_direct_mentioned = (row["flags"] & UserMessage.flags.mentioned_me_directly) != 0
         if is_mentioned:
             mentions.add(message_id)
+        if is_direct_mentioned:
+            mentions_me_directly.add(message_id)
```

Comment:
according to [this error](https://github.com/zulip/zulip/actions/runs/6632305109/job/18017687171?pr=27306#step:16:1589), `ValueError: 'mentioned_me_directly' is not in list`, it seems like `mentioned_me_directly` isn't a `flag` they for `UserMessage`

looks like all the flags are [here in this file](https://github.com/zulip/zulip/blob/ba568adc2e4e08842c028ed0396d05d99533527e/zerver/models.py#L3505)

I'm assuming that, in the DB, just `mentioned` is used to save direct mentions, since `wildcard_mentioned` is still its own field

i think the easiest way to avoid confusion here would be to rename the boolean `is_mentioned` to `is_direct_mention` (even tho the flag name is `mentioned`), and append to both the generic `mentioned` set and the more specific `mentions_me_directly` set

```suggestion
        is_direct_mention = (row["flags"] & UserMessage.flags.mentioned) != 0
        is_wildcard_mentioned = (row["flags"] & UserMessage.flags.wildcard_mentioned) != 0
        if is_direct_mention:
            mentions.add(message_id)
            mentions_me_directly.add(message_id)
```